src/demo/components/Graphs.tsx

```diff
@@ -0,0 +1,230 @@
+import '../demo.css'
+
+import React from 'react'
+import {
+  Bar,
+  CartesianGrid,
+  ComposedChart,
+  Line,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis
+} from 'recharts'
+
+interface Bucket {
+  start: number
+  usdValue: number
+  numTxs: number
+  isoDate: string
+  currencyCodes: { [currencyCode: string]: number }
+  currencyPairs: { [currencyPair: string]: number }
+}
+
+interface AnalyticsResult {
+  result: {
+    hour: Bucket[]
+    day: Bucket[]
+    month: Bucket[]
+    numAllTxs: number
+  }
+  app: string
+  pluginId: string
+  start: number
+  end: number
+}
+
+const Graphs: any = (props: {
+  rawData: AnalyticsResult[]
+  timePeriod: string
+  colors: string[]
+}) => {
+  const { rawData, timePeriod } = props
+  const keys = props.rawData.map(analyticsResult => {
+    return (
+      analyticsResult.pluginId.charAt(0).toUpperCase() +
+      analyticsResult.pluginId.slice(1)
+    )
+  })
+  const colors = {}
+  for (let i = 0; i < rawData.length; i++) {
+    colors[keys[i]] = props.colors[i]
+  }
+  const tickRate: string[] = []
+  let tickSpace = 0
+  tickSpace = Math.floor(rawData[0].result[timePeriod].length / 5)
+  if (tickSpace === 0) tickSpace++
+
+  const data = rawData[0].result[timePeriod].map((bucket: Bucket, index) => {
+    let date
+    if (timePeriod === 'month') {
+      date = bucket.isoDate.slice(0, 7)
+    } else if (timePeriod === 'day') {
+      date = bucket.isoDate.slice(0, 10)
+    } else {
+      date = bucket.isoDate.slice(0, 10) + ':' + bucket.isoDate.slice(11, 13)
+    }
+    if (index % tickSpace === 0) {
+      tickRate.push(date)
+    }
+    const currencyPairs: { [currencyPair: string]: number } = {}
+    for (const bucket of rawData) {
+      for (const [key, value] of Object.entries(
+        bucket.result[timePeriod][index].currencyPairs
+      )) {
+        if (typeof value === 'number') {
+          if (currencyPairs[key] == null) {
+            currencyPairs[key] = value
+            continue
+          }
+          currencyPairs[key] += value
+        }
+      }
+    }
+    const currencyPairArray = Object.entries(currencyPairs).sort(
+      (a, b) => b[1] - a[1]
+    )
+    return { date, allUsd: 0, allTxs: 0, currencyPairs: currencyPairArray }
+  })
+  for (let i = 0; i < rawData.length; i++) {
```

Comment:
for (let i = 0; i < rawData.length; i++) {
    const analytic = rawData[i]
    const txData = analytic.result[timePeriod]
    for (let j = 0; j < txData.length; j++) {
      const { usdValue, numTxs } = txData[j]
      const graphName =
        analytic.pluginId.charAt(0).toUpperCase() + analytic.pluginId.slice(1)
      data[j].allUsd += usdValue
      data[j][graphName] = usdValue
      data[j].allTxs += numTxs
      data[j][`${graphName}NumTxs`] = numTxs
      data[j][`${graphName}Color`] = props.colors[i]
    }
  }